packages/shared/src/components/modals/streaks/NewStreakModal.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { ReactElement, useState } from 'react';
+import classNames from 'classnames';
+import { LazyModalCommonProps, Modal } from '../common/Modal';
+import FireIcon from '../../icons/Fire';
+import classed from '../../../lib/classed';
+import TwitterIcon from '../../icons/Twitter';
+import { ButtonVariant } from '../../buttons/common';
+import { ButtonColor } from '../../buttons/ButtonV2';
+import { SocialShareButton } from '../../widgets/SocialShareButton';
+import WhatsappIcon from '../../icons/Whatsapp';
+import StraightArrowIcon from '../../icons/Arrow/Straight';
+import { IconSize } from '../../Icon';
+import { Checkbox } from '../../fields/Checkbox';
+import { useActions } from '../../../hooks';
+import { ActionType } from '../../../graphql/actions';
+import { ModalClose } from '../common/ModalClose';
+import { SplashIcon } from '../../icons/Splash';
+
+interface FirstStreakModalProps extends LazyModalCommonProps {
+  currentStreak: number;
+  previousStreak: number;
+}
+
+const Paragraph = classed('p', 'text-center text-theme-label-tertiary');
+
+export default function FirstStreakModal({
+  currentStreak,
+  previousStreak,
+  onRequestClose,
+  ...props
+}: FirstStreakModalProps): ReactElement {
+  const [isChecked, setIsChecked] = useState(false);
+  const { completeAction } = useActions();
+  const shouldShowSplash = currentStreak > 20;
+  const onClose = (e: React.MouseEvent | React.KeyboardEvent) => {
+    if (isChecked) {
+      completeAction(ActionType.OptOutStreaks);
+    }
+
+    onRequestClose(e);
+  };
+
+  return (
+    <Modal
+      {...props}
+      kind={Modal.Kind.FlexibleTop}
+      size={Modal.Size.XSmall}
+      onRequestClose={onClose}
+    >
+      <ModalClose onClick={onRequestClose} className="right-2 top-2" />
+      <Modal.Body className="items-center">
+        <span className="relative flex flex-col items-center justify-center">
+          {shouldShowSplash ? (
+            <SplashIcon className="ml-2 h-[10rem] w-[15rem] text-theme-color-bacon" />
+          ) : (
+            <FireIcon className="h-[10rem] w-[10rem] text-theme-color-bacon" />
```

Comment:
Not sure why `size-[10rem]` doesn't seem to work here.

Reply:
Hmm arbitrary might need the removal of `.next` and a re-build? (not 100% sure)

Reply:
Don't we have icon sizes like Small, medium.. we can use?

Reply:
Moved to an image element.